src/main/java/com/scalar/cassy/util/AzureUtil.java

```diff
@@ -3,12 +3,38 @@
 import com.azure.identity.DefaultAzureCredentialBuilder;
 import com.azure.storage.blob.BlobContainerAsyncClient;
 import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
 import com.azure.storage.blob.BlobServiceClientBuilder;
 
 public class AzureUtil {
   private static final String CONNECTION_STRING = "AZURE_STORAGE_CONNECTION_STRING";
 
   public static BlobContainerAsyncClient getBlobContainerAsyncClient(String storageBaseUri) {
+    BlobServiceClientBuilder builder = getBlobServiceClientBuilder(storageBaseUri);
+    BlobServiceAsyncClient asyncClient = builder.buildAsyncClient();
+
+    if (!storageBaseUri.startsWith(asyncClient.getAccountUrl())) {
```

Comment:
This if statement is a bit redundant so why don't you create a method like this?
```
private void validateUrl(String accountUrl) {
  if (!storageBaseUri.startsWith(accountUrl)) {     
    throw new IllegalArgumentException(              
    "The given credential can not be used for the specified container.");   
  }
}
```

Reply:
Sorry for the inconvenience.
Implementation modified for avoiding redundancy.
```
  private static void validateUrl(String storageBaseUri, String accountUrl) {
    if (!storageBaseUri.startsWith(accountUrl)) {
      throw new IllegalArgumentException(
              "The given credential can not be used for the specified container.");
    }
  }
```